orttraining/orttraining/python/training/ortmodule/_training_manager.py

```diff
@@ -41,8 +41,11 @@ def execution_session_run_forward(execution_session, onnx_model, *inputs):
 
         forward_outputs = C.OrtValueVector()
         # Run and return module outputs.
-        execution_session.run_forward(forward_inputs, forward_outputs, state)
-        user_outputs = tuple(_utils._ortvalue_to_torch_tensor(forward_output) for forward_output in forward_outputs)
+        execution_session.run_forward(forward_inputs, forward_outputs, state, cache)
+        user_outputs = tuple(_utils._ortvalue_to_torch_tensor(forward_outputs[i]) for i in range(cache_start))
+        if cache != None:
+            for i in range(cache_start, len(forward_outputs)):
+                cache.insert(cache_names[i-cache_start], forward_outputs[i])
```

Comment:
this will internally call https://www.cplusplus.com/reference/unordered_map/unordered_map/emplace/
see https://github.com/microsoft/onnxruntime/blob/90ef8f07eb158fe0133f39ecc2b7b0ab11941276/onnxruntime/core/framework/partial_graph_execution_state.h#L29